core/src/main/scala/org/http4s/Query.scala

```diff
@@ -133,14 +135,22 @@ final class Query private (value: Either[Vector[KeyValue], String])
     * none exist, the empty `String` "" is returned.
     */
   lazy val params: Map[String, String] =
-    CollectionCompat.mapValues(multiParams)(_.headOption.getOrElse(""))
+    multiParams.view.mapValues(_.headOption.getOrElse("")).toMap
 
   /** Map[String, Seq[String]] representation of the [[Query]]
     *
     * Params are represented as a `Seq[String]` and may be empty.
     */
   lazy val multiParams: Map[String, immutable.Seq[String]] =
-    CollectionCompat.pairsToMultiParams(toVector)
+    if (toVector.isEmpty) Map.empty
+    else {
+      val m = mutable.Map.empty[String, ListBuffer[String]]
```

Comment:
I see this already existed before this PR, but does anyone know why we are using a mutable Map here?

Reply:
It might be faster and I don't think the immutability is not observable.

Reply:
> It might be faster and I don't think the immutability is not observable.

Yeah, I thought faster might be the reason, though I'm a bit skeptical of that unless we showed this was hot path and/or profiled it. 

Not a big deal though, it is totally hidden and easy enough to validate that it is correct.

Reply:
I suspect the performance is similar enough, and the code local enough, it really makes little difference for performance or maintainability either way.  But I have open source YourKit licenses for anyone who wants to do Real Science.

I'm going to start beating on things with a profiler myself... after the Dotty and CE3 projects are done. :smile: